server/src/graphql/api.ts

```diff
@@ -72,15 +62,17 @@ export const graphqlRoot: Resolvers<Context> = {
         })
         .limit(25)
         .getMany()
-      if (checkSeen) {
-        // This statement requires a semicolon
-        const seenArt: Set<number> = new Set();
-        (await User.findOne())?.artSeen.forEach(art => { seenArt.add(art.id) })
-        result.forEach((art: any) => {
-          art.seen = seenArt.has(art.id)
-        })
-      }
-      return result
+      // if (checkSeen) {
+      //   // This statement requires a semicolon
+      //   const seenArt: Set<number> = new Set()
+      //   ;(await User.findOne())?.artSeen.forEach(art => {
+      //     seenArt.add(art.id)
+      //   })
+      //   result.forEach((art: any) => {
+      //     art.seen = seenArt.has(art.id)
+      //   })
+      // }
+      return result as any
```

Comment:
It occurs to me that we have `seen` as a boolean field on art on the API level. I think the intent was that this was populated using the user context of the request (so we don't get it for free from the DB) so we can color the points on the map. Do we still want to do this, or simply grab the user (and its seen art) on the front end and process it there?

Reply:
If it was the way before, I'd only need the single query to fetch nearby artwork and check the `seen` boolean flag for map coloring, but I think with the way it is now I'll have to do queries for both artworkSeen and nearby, then filter the artwork list in the frontend. I think it'll be more efficient the first way, but lmk if you disagree

Reply:
In the DB, I guess we could add a seenBy field on art, then just use that? I'm kind of surprised that I didn't put the field in previously.

Reply:
I got rid of this because the idea of passing a parameter and querying GraphQL that way is definitely an anti-pattern. I guess the best way to deal with this in an efficient manner would be to write our query to get a single piece of artwork just
```sql
SELECT * FROM art WHERE id = '1';
```
Then we can have separate resolver for the `seenByUser` and `likedByUser` fields do something like
```sql
SELECT 1 FROM artSeenJoinTable WHERE userid = x AND artid = y;
```
That could be efficiently rewritten into a dataloader-type request like
```sql
SELECT id, CASE WHEN artid IS NULL FALSE ELSE TRUE
FROM artid
LEFT JOIN artSeenTable t ON t.artid = id 
WHERE artid IN (...) AND userid = x;
```